marker_rustc_driver/src/conversion/marker/expr.rs

```diff
@@ -392,6 +411,54 @@ impl<'ast, 'tcx> MarkerConverterInner<'ast, 'tcx> {
         )
     }
 
+    fn to_expr_from_closure(
+        &self,
+        data: CommonExprData<'ast>,
+        _expr: &hir::Expr<'tcx>,
+        closure: &hir::Closure<'tcx>,
+    ) -> ExprKind<'ast> {
+        let body_id = closure.body;
+        let body = self.rustc_cx.hir().body(body_id);
+        match body.generator_kind {
+            Some(hir::GeneratorKind::Async(hir::AsyncGeneratorKind::Fn)) => {
+                if let hir::ExprKind::Block(block, None) = body.value.kind
+                    && let Some(temp_drop) = block.expr
+                    && let hir::ExprKind::DropTemps(inner_block) = temp_drop.kind
+                {
+                    let api_expr;
+                    super::with_body!(self, body_id, {
```

Comment:
The `with_body` was a quick fix when I added it, I can see if a function taking a closure, will work equally well and otherwise document it accordingly